File: java/IEC61970/Dynamics/StandardModels/VoltageAdjusterDynamics/VoltageAdjusterDynamics.java
package IEC61970.Dynamics.StandardModels.VoltageAdjusterDynamics;

import IEC61970.Dynamics.StandardModels.DynamicsFunctionBlock;
import IEC61970.Dynamics.StandardModels.PFVArControllerType1Dynamics.PFVArControllerType1Dynamics;

/**
 * Voltage adjuster function block whose behaviour is described by reference to a
 * standard model <font color="#0f0f0f">or by definition of a user-defined model.
 * </font>
 * @author ppbr003
 * @version 1.0
 * @created 25-Dec-2023 8:32:04 PM
 */
public class VoltageAdjusterDynamics extends DynamicsFunctionBlock {

	/**
	 * Power Factor or VAr controller Type I model with which this voltage adjuster is
	 * associated.
	 */
	public PFVArControllerType1Dynamics PFVArControllerType1Dynamics;

	public VoltageAdjusterDynamics(){

	}
}//end VoltageAdjusterDynamics

File: py/IEC61970/Dynamics/StandardModels/VoltageAdjusterDynamics/VoltageAdjusterDynamics.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106 on Sun Dec 31 17:26:32 2023
from datetime import datetime
from typing import Optional

from IEC61970.Dynamics.StandardModels.DynamicsFunctionBlock import DynamicsFunctionBlock
from IEC61970.Dynamics.StandardModels.PFVArControllerType1Dynamics.PFVArControllerType1Dynamics import \
    PFVarControllerType1Dynamics


class VoltageAdjusterDynamics(DynamicsFunctionBlock):
    """
    Voltage adjuster function block whose behaviour is described by reference to a
    standard model or by definition of a user-defined model.
    @author ppbr003
    @version 1.0
    @created 29-Dec-2023 11:24:20 PM
    """

    def __init__(self) -> None:
        super().__init__()
        # Power Factor or VAr controller Type I model with which this voltage adjuster is associated.
        self.pf_var_controller_type1_dynamics: PFVarControllerType1Dynamics = PFVarControllerType1Dynamics()
